scipy/optimize/_numdiff.py

```diff
@@ -344,34 +365,81 @@ def fun_wrapped(x):
     if np.any((x0 < lb) | (x0 > ub)):
         raise ValueError("`x0` violates bound constraints.")
 
-    h = _compute_absolute_step(rel_step, x0, method)
+    if as_linear_operator:
+        if rel_step is None:
+            rel_step = _compute_relative_step(method)
+
+        return _linear_operator_difference(fun_wrapped, x0,
```

Comment:
Just thought about that: why we should use `rel_step` here?

A conventional error analysis (in a scalar case) for a forward difference gives us that the "optimal" step is given by: `p * h = sqrt(eps * f / f'')` (see Numerical Recipes Third Edition p. 230). I assumed that roundoff error comes only from numerator (as in the book), which can be violated here. 

And usually `sqrt(f/f'')` is replaced by `x` (This is something to think sceptically about, but nevertheless this is how it is done currently when `as_linear_operator=False`). 

And you decided to use `h = sqrt(eps)`, which frankly I don't think is worse than any other "smarter" choice, but still needs some considerations. Are you sure it can't be improved or turned to a conventional approach?